CONTRIBUTING.md

```diff
@@ -1,16 +1,225 @@
+
+* Each commit should target changes of specific parts/modules of RIOT. The
+  commits use the following pattern:
+  `path/to/directory: description of changes`.
+  You can use multi-line commit messages if you want to detail more the
+  changes.
+
+* Try to answer reviews as quickly as possible to speed up the review process
+  and avoid staled PRs.
+
+You can find more information about RIOT development procedure on this
+[wiki page](https://github.com/RIOT-OS/RIOT/wiki/Development-procedures).
+
+### Git usage
+[git-usage]: #git-usage
+
+Using git is a bit difficult for newcomers. In this section, we give the bare
+minimum for a better experience with our development workflow on GitHub.
+
+#### Setup your local RIOT repository
+
+Before you start modifying code, you need to fork the RIOT upstream repository
+from the [RIOT main github page](https://github.com/RIOT-OS/RIOT).
+
+Then clone locally your fork of RIOT (replace `account name` with your actual
+login on GitHub):
+
+    $ git clone git@github.com:<account name>/RIOT.git
+
+You can keep any branch of your local repository up-to-date with the upstream
+master branch with the following commands:
+
+    $ git checkout <branch name>
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+
+Use it at least before opening a PR. This will ensure the PR can be merged
+
+#### Work on branches
+
+Avoid opening PR from the `master` branch of your fork to the master branch of
+the RIOT upstream repository: update your master branch and start a new branch
+from it:
+
+    $ git checkout master
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+    $ git checkout -b <new branch>
+    # Do your changes, commit, update with latest upstream master
+    $ git push
+
+#### Add fixup commits
+
+To keep the history of changes easier to track for reviewers, it is recommended
+to push your review request updates in fixup commits.
+
+Let's say your PR contains 3 commits with comments: `dir1: change 1`,
+`dir2: change 2` and `dir3: change 3`.
+
+Instead of commiting changes in dir2 in a 4th commit `dir2: change 4`, you can
+use the `--fixup` option:
+
+    $ git add dir2
+    $ git commit --fixup <dir2 commit hash>
+
+#### Squash commits
+
+**Don't squash your commit until a maintainer asked you for doing it.**
+
+Squashing commit is done using the rebase command of git in interactive mode:
+
+    $ git rebase master -i
+
+You can find information on rebasing on this [Github page][about-git-rebase].
+
+[about-git-rebase]: https://help.github.com/articles/about-git-rebase/
+
+If you used fixup commits during the review phase, squashing commits can be
+performed in a single command:
+
+    $ git rebase -i --autosquash
+
+## Writing Documentation
+[writing-documentation]: #writing-documentation
+
+Documentation improvements are always welcome and a good starting point for
+new contributors. This kind of contributions are merged quite fast in general.
```

Comment:
I think it should be either "These kind of contributions are" or "This kind of contribution is"